wasmbinding/queries.go

```diff
@@ -124,6 +130,82 @@ func (qp *QueryPlugin) GetFailures(ctx sdk.Context, address string, pagination *
 	return &bindings.FailuresResponse{Failures: res.Failures}, nil
 }
 
+func dexQuery[T any, R any](ctx sdk.Context, query *T, queryHandler func(ctx context.Context, query *T) (R, error)) ([]byte, error) {
+	resp, err := queryHandler(ctx, query)
+	if err != nil {
+		return nil, errors.Wrapf(err, fmt.Sprintf("failed to query request %T", query))
+	}
+
+	data, err := json.Marshal(resp)
+	if err != nil {
+		return nil, errors.Wrapf(err, fmt.Sprintf("failed to marshal response %T", resp))
+	}
+
+	return data, nil
+}
+
+func (qp *QueryPlugin) DexQuery(ctx sdk.Context, query bindings.DexQuery) ([]byte, error) {
+	var err error
+	var data []byte
+	switch {
+	case query.EstimateMultiHopSwap != nil:
+		data, err = dexQuery(ctx, query.EstimateMultiHopSwap, qp.dexKeeper.EstimateMultiHopSwap)
+	case query.EstimatePlaceLimitOrder != nil:
+		q := dextypes.QueryEstimatePlaceLimitOrderRequest{
+			Creator:          query.EstimatePlaceLimitOrder.Creator,
+			Receiver:         query.EstimatePlaceLimitOrder.Receiver,
+			TokenIn:          query.EstimatePlaceLimitOrder.TokenIn,
+			TokenOut:         query.EstimatePlaceLimitOrder.TokenOut,
+			TickIndexInToOut: query.EstimatePlaceLimitOrder.TickIndexInToOut,
+			AmountIn:         query.EstimatePlaceLimitOrder.AmountIn,
+			OrderType:        query.EstimatePlaceLimitOrder.OrderType,
+			MaxAmountOut:     query.EstimatePlaceLimitOrder.MaxAmountOut,
+		}
+		if query.EstimatePlaceLimitOrder.ExpirationTime != nil {
+			t := time.Unix(int64(*query.EstimatePlaceLimitOrder.ExpirationTime), 0)
```

Comment:
actually it can't, but int64 overflow happens when number > 9223372036854775807, which is quite significant in seconds, overflow gives you negative number.  int64(9223372036854775807 + 1) == -9223372036854775808 
Anyway, i dont think someone wants to set expiration time way after our universe dies.